pyccel/ast/core.py

```diff
@@ -2070,14 +2083,37 @@ def __init__(self, func, args, current_function=None):
             if len(func.results)>0 and not isinstance(func.results[0], TypedAstNode):
                 errors.report(RECURSIVE_RESULTS_REQUIRED, symbol=func, severity="fatal")
 
-        self._funcdef       = func
-        self._arguments     = args
-        self._dtype         = func.results[0].var.dtype     if len(func.results) == 1 else NativeTuple()
-        self._rank          = func.results[0].var.rank      if len(func.results) == 1 else None
-        self._shape         = func.results[0].var.shape     if len(func.results) == 1 else None
-        self._precision     = func.results[0].var.precision if len(func.results) == 1 else None
-        self._order         = func.results[0].var.order     if len(func.results) == 1 else None
-        self._func_name     = func.name
+        self._funcdef    = func
+        self._arguments  = args
+        self._func_name  = func.name
+        n_results = len(func.results)
+        if n_results == 1:
+            self._dtype      = func.results[0].var.dtype
+            self._rank       = func.results[0].var.rank
+            self._shape      = func.results[0].var.shape
+            self._precision  = func.results[0].var.precision
+            self._order      = func.results[0].var.order
+            self._class_type = func.results[0].var.class_type
+        elif n_results == 0:
+            self._dtype      = NativeGeneric()
+            self._rank       = 1
+            self._shape      = (LiteralInteger(0),)
+            self._precision  = None
+            self._order      = None
+            self._class_type = NativeTuple()
```

Comment:
I cannot understand these changes. In Python a function with no returns is a function which returns `None`, so how can this be a tuple of shape `(0,)`?

Reply:
I think I must have been thinking of an empty tuple when I wrote this. Fixed [1961ed9](https://github.com/pyccel/pyccel/pull/1590/commits/1961ed986f719fe59fd15599ac8a1b002260a1f6)